### PR TITLE
Assignment7 : mmwoodfo

### PR DESCRIPTION
**Pull request by mmwoodfo**

### CODE REVIEW

File: DelegateFun/Sorter.Tests/SortUtilityTests.cs [hunk 1]
```diff
@@ -1,13 +1,129 @@
...
     {
         [TestMethod]
-        public void SortUtility_ShouldSortAscending_UsingAnAnonymousMethod()
+        public void QuickSort_NullArray_ThrowNullException()
         {
+            //Arrange
+            SortUtility sorter = new SortUtility();
+
+            //Act & Assert
+            Assert.ThrowsException<ArgumentNullException>(() => sorter.QuickSort(null!, ((int num1, int num2) => num1 < num2), 0, 0));
```
Comment (dsergio) [inline DelegateFun/Sorter.Tests/SortUtilityTests.cs]: nice

File: DelegateFun/Sorter.Tests/SortUtilityTests.cs [hunk 2]
```diff
@@ -1,13 +1,129 @@
...
+            Compare compareNumbers = delegate (int num1, int num2)
+            {
+                return num1 < num2;
+            };
+
+            //Act
+            sorter.QuickSort(intArray, compareNumbers, 0, intArray.Length - 1);
+
+            //Assert
+            CollectionAssert.AreEqual(expectedIntArray, intArray);
```
Comment (dsergio) [inline DelegateFun/Sorter.Tests/SortUtilityTests.cs]: CollectionAssert, nice

File: DelegateFun/Sorter/SortUtility.cs [hunk 3]
```diff
@@ -2,9 +2,49 @@
...
+            }
+
+            int temp2 = intArray[smaller + 1];
+            intArray[smaller + 1] = intArray[high];
+            intArray[high] = temp2;
+
+            return smaller + 1;
+        }
+
+        public void QuickSort(int[] intArray, Compare comparer, int low, int high)
```
Comment (dsergio) [inline DelegateFun/Sorter/SortUtility.cs]: might be better to make this private and send 0 and intArray.Length as parameters in a public method

Comment (mmwoodfo) [reply DelegateFun/Sorter/SortUtility.cs]: I was considering that. I may make that change now that someone else has suggested it. :) Thank you!

**Pull request closed without merging**